src/plugins/data_source/server/auth_registry/authentication_methods_registry.ts

```diff
@@ -18,6 +19,15 @@ export class AuthenticationMethodRegistery {
    * Authentication Method can only be registered once. subsequent calls with the same method name will throw an error.
    */
   public registerAuthenticationMethod(method: AuthenticationMethod) {
+    if (
+      method.name === AuthType.NoAuth ||
+      method.name === AuthType.UsernamePasswordType ||
+      method.name === AuthType.SigV4
+    ) {
+      throw new Error(
```

Comment:
will this block auth types being registered by other plugins?

Reply:
This condition does not allow auth types to be registered with the reserved auth type names. So that it won't accidentally interfere with our existing auth types.